### ISSUE TITLE
Functional tests failing due to setup function skip.

### ISSUE THREAD

**Issue by dashmage** [canonical/hardware-observer-operator#122]
Running `make functional31` skips the `test_build_and_deploy` setup function and causes the rest of the tests to fail. Here's the relevant bits from the error log

```
tests/functional/test_charm.py::test_build_and_deploy
  tests/functional/test_charm.py:43: PytestWarning: The test <Function test_build_and_deploy> is marked with '@pytest.mark.asyncio' but it is not an async function. Please remove the asyncio mark. If the test is not marked explicitly, check for global marks applied via 'pytestmark'.
    @pytest.mark.abort_on_fail

tests/functional/test_charm.py::test_build_and_deploy
  /home/dashmage/Canonical/repo/hardware-observer-operator/.tox/func31/lib/python3.11/site-packages/_pytest/python.py:183: PytestUnhandledCoroutineWarning: async def functions are not natively supported and have been skipped.
  You need to install a suitable plugin for your async framework, for example:
    - anyio
    - pytest-asyncio
    - pytest-tornasync
    - pytest-trio
    - pytest-twisted
    warnings.warn(PytestUnhandledCoroutineWarning(msg.format(nodeid)))
```

Error can be seen in [this failed CI run](https://github.com/canonical/hardware-observer-operator/actions/runs/7061188853/job/19272294383). On GitHub, the test is passing for `make functional31` which isn't the case on my local.

**Comment by dashmage** [comment]
[Working](https://github.com/canonical/hardware-observer-operator/actions/runs/7083136244/job/19275048570?pr=121) with changes in [this commit](https://github.com/canonical/hardware-observer-operator/pull/121/commits/36be58604e5dc5e6b5736a94e4595df6eccb84b1). 

Possible source of error:
`pytest-asyncio` has some breaking changes in [v0.23.0 released on 12/03](https://pytest-asyncio.readthedocs.io/en/latest/reference/changelog.html#id3) and is used by `pytest-operator` internally. 

[This issue](https://github.com/pytest-dev/pytest-asyncio/issues/701) where it's also causing some unneccessary skips, has been fixed in `v0.23.1`. But even after using this latest version, I'm not able to get the func tests to work without adding `pytestmark = pytest.mark.asyncio(scope="module")` in the test.

**Comment by jneo8** [comment]
https://github.com/charmed-kubernetes/pytest-operator/pull/120

Fix is there. We should be fine now.
